### DISCUSSION TITLE
tailwind.config.js - blocklist

### DISCUSSION THREAD

**Discussion by K-Thompson** [tailwindlabs/tailwindcss#10151] [Help] [answered]
I'm using Tailwindcss 3.2 and Bootstrap 5. Both have a `.collapse` class which is incompatible with each other.

I'm familiar with the `blocklist` feature, but this feature prevents the class from being created regardless of its source. I want to prevent Tailwindcss from generating the class and retain the class definition from Bootstrap.

Is there a way to do this? Or how does a developer handle class conflicts?

See [this discussion](https://github.com/tailwindlabs/tailwindcss/issues/330) about prefixing specific classes from Dec. 20, 2017. At one time, there was the ability to do this, but I gather it is no longer supported.

**Comment by wongjn** [comment, marked as answer]
As per [the `blocklist` documentation](https://tailwindcss.com/docs/content-configuration#discarding-classes):

> The `blocklist` option only affects CSS that would be generated by Tailwind, not custom CSS you’ve authored yourself or are importing from another library.

So if you find that any `.collapse` rules from another library are not present in your CSS, it could be some misconfiguration, but they *should* be present.

**Reply by K-Thompson** [reply]
Thank you, @wongjn, for pointing this out. You are right; I had my custom CSS misconfigured. The Bootstrap `.collapse` was within `@layer components`. Moving it outside the `@layer` fixed the problem for me.

**Reply by ProPhycient5** [reply]
@K-Thompson, please close the discussion by marking @wongjn reply as `answer`.
